obspy/clients/fdsn/mass_downloader/restrictions.py

```diff
@@ -199,7 +199,8 @@ def __init__(self, starttime, endtime,
                                      "MH[ZNE12]", "EH[ZNE12]",
                                      "LH[ZNE12]", "HL[ZNE12]",
                                      "BL[ZNE12]", "ML[ZNE12]",
-                                     "EL[ZNE12]", "LL[ZNE12]"),
+                                     "EL[ZNE12]", "LL[ZNE12]",
+                                     "SH[ZNE12]"),
                  location_priorities=("", "00", "10")):
```

Comment:
I think `location_priorities` should be extended to contain more values.

Location codes are usually random. People wouldn't known the exact location codes unless they request station metadata before requesting waveform data. Thus, most people, I presume, may prefer to leave the `location` parameter to `None`. In this case, `location_priorities` will determine which data should be downloaded. 

Since location codes in current default `location_priorities` are incomplete, some data will be missed unexpectedly. For example, high gain (`?H?`) channels of `YA` network have location codes of `01` and `02`. If I use the default `location` and `location_priorities`, I will get no data. See the link below for details about `YA` network.

http://service.iris.edu/fdsnws/station/1/query?net=YA&cha=?H?&level=channel&format=text&maxlat=54&minlon=65&maxlon=135&minlat=18&includecomments=true&nodata=404

My suggestion:
1. add more common value to `location_priorities`, e.g., 01, 02, 06, 60...
2. and wildcards `[0-9][0-9]`, `[0-9][A-Z]` and `[A-Z][0-9]` as the last three items of `location_priorities`, to make sure that at least some data are downloaded if the channel codes fail to match other codes in `location_priorities`.

One reference which may be useful:
1. https://ds.iris.edu/ds/newsletter/vol1/no1/specification-of-seismograms-the-location-identifier/